File: 283.Move Zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int zeroIndex = 0;
        for (int i=0; i<nums.size(); i++) {
            if (nums[i] != 0) {
                nums[zeroIndex] = nums[i];
                zeroIndex++;
            } 
        }
        
        for (int i=zeroIndex; i<nums.size(); i++) {
            nums[i] = 0;
        }
    }
};

File: LeetCode/283.Move Zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        i = 0
        while i < len(nums) and nums[i] != 0:
            i += 1
        j = i + 1
        while j < len(nums):
            if nums[j] != 0:
                nums[i] = nums[j]
                i += 1
            j += 1
        while i < len(nums):
            nums[i] = 0
            i += 1
